bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/DahuaHandler.java

```diff
@@ -169,18 +169,94 @@ private void processEvent(String content) {
                     ipCameraHandler.setChannelState(CHANNEL_TOO_BLURRY_ALARM, OnOffType.OFF);
                 }
                 break;
+            case "AccessControl":
+                if ("Pulse".equals(action)) {
+                    if (content.contains("\"Method\" : 1")) {
+                        startIndex = content.indexOf("CardNo", startIndex) + 11;
+                        if (startIndex > 0) {
+                            endIndex = content.indexOf(",", startIndex) - 1;
+                            String cardNo = content.substring(startIndex, endIndex);
+                            ipCameraHandler.setChannelState(CHANNEL_ACCEPTED_CARD_NUMBER, new StringType(cardNo));
+                            ipCameraHandler.setChannelState(CHANNEL_DOOR_UNLOCK, OnOffType.ON);
+                        }
+                    }
+                    if (content.contains("\"Method\" : 5")) {
+                        ipCameraHandler.setChannelState(CHANNEL_DOOR_UNLOCK, OnOffType.ON);
+                        ipCameraHandler.logger.debug("Door opened from button");
+                    }
+                    if (content.contains("\"Method\" : 4")) {
+                        ipCameraHandler.setChannelState(CHANNEL_DOOR_UNLOCK, OnOffType.ON);
+                        ipCameraHandler.logger.debug("Door opened remotely");
+                    }
+                } else {
+                    ipCameraHandler.logger.debug("Unrecognised Access control Dahua event, content={}", content);
+                }
+                break;
+            case "DoorCard":
+                if ("Pulse".equals(action)) {
+                    if (content.contains("\"Number\"")) {
+                        startIndex = content.indexOf("Number", startIndex) + 11;
+                        if (startIndex > 0) {
+                            endIndex = content.indexOf(",", startIndex) - 1;
+                            String cardNo = content.substring(startIndex, endIndex);
+                            ipCameraHandler.logger.debug("Door recognized an access card, {}", cardNo);
```

Comment:
Is this a bug? The code takes the time to get cardNo, then does nothing with it except to use it in the logs?

Reply:
Again - that`s a debug log, invisible in general, and as i mentioned [above](https://github.com/openhab/openhab-addons/pull/13313#issuecomment-1273310379), that's an alternative way to deal with access cards. It could be added later as a channel for those, who want openHAB to deal with cards.

Reply:
The general rule is to not use the logs as a debugger, so my opinion is that the binding should be as low impact on a system as possible if there is nothing to be gained. Since this only runs when a card is scanned and not regularly, I do not feel strongly about it, so please let me know if you wish to leave it as is as I have already reviewed it and you have already written it.... In future, please finish a feature fully as it increases lines of code that need to be maintained and more work for reviewers which we need more of.
The load on the system is not just logging, its searching strings, creating sub strings and checking what level the logger is set to as it gives no benefit in return, only a load which is why I have commented on it.